subsys/bluetooth/audio/ascs.c

```diff
@@ -285,198 +398,143 @@ void ascs_ep_set_state(struct bt_bap_ep *ep, uint8_t state)
 		}
 	}
 
-	if (stream->ops != NULL) {
-		const struct bt_bap_stream_ops *ops = stream->ops;
-
-		switch (state) {
-		case BT_BAP_EP_STATE_IDLE:
-			ep->receiver_ready = false;
-			bt_bap_stream_reset(stream);
-
-			if (ops->released != NULL) {
-				ops->released(stream);
-			}
-			struct bt_ascs_ase *ase = CONTAINER_OF(ep, struct bt_ascs_ase, ep);
+	switch (new_state) {
+	case BT_BAP_EP_STATE_IDLE:
+		ase_set_state_idle(ase);
+		break;
+	case BT_BAP_EP_STATE_CODEC_CONFIGURED:
+		ase_set_state_codec_configured(ase);
+		break;
+	case BT_BAP_EP_STATE_QOS_CONFIGURED:
+		ase_set_state_qos_configured(ase);
+		break;
+	case BT_BAP_EP_STATE_ENABLING:
+		ase_set_state_enabling(ase);
+		break;
+	case BT_BAP_EP_STATE_STREAMING:
+		ase_set_state_streaming(ase);
+		break;
+	case BT_BAP_EP_STATE_DISABLING:
+		ase_set_state_disabling(ase);
+		break;
+	case BT_BAP_EP_STATE_RELEASING:
+		ase_set_state_releasing(ase);
+		break;
+	default:
+		__ASSERT_PRINT("Invalid state %d", new_state);
+	}
```

Comment:
This does make it significantly easier to read - We should consider applying the same for the other BAP state machines